### PR TITLE
Add pydamage analyze module

### PR DESCRIPTION
**Pull request by maxibor**
## PR checklist

Adding the [PyDamage](https://github.com/maxibor/pydamage) module

- [x] This comment contains a description of changes (with reason).
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] Remove all TODO statements.
- [x] Emit the `<SOFTWARE>.version.txt` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
    - [x] `PROFILE=docker pytest --tag pydamage --symlink --keep-workflow-wd`
    - [] `PROFILE=conda pytest --tag pydamage --symlink --keep-workflow-wd`

### CODE REVIEW

File: modules/pydamage/main.nf [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+options        = initOptions(params.options)
+
+process PYDAMAGE {
+    tag "$meta.id"
+    label 'process_medium'
+    publishDir "${params.outdir}",
+        mode: params.publish_dir_mode,
+        saveAs: { filename -> saveFiles(filename:filename, options:params.options, publish_dir:getSoftwareName(task.process), meta:meta, publish_by_meta:['id']) }
+
+        conda (params.enable_conda ? "bioconda::pydamage=0.61" : null)
```
Comment (jfy133) [inline modules/pydamage/main.nf:15]:
```suggestion
        conda (params.enable_conda ? "bioconda::pydamage=0.62" : null)
```

File: modules/pydamage/main.nf [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        container "https://depot.galaxyproject.org/singularity/pydamage:0.62--pyhdfd78af_0"
+    } else {
+        container "quay.io/biocontainers/pydamage:0.62--pyhdfd78af_0"
+    }
+
+    input:
+    tuple val(meta), path(bam), path(bai)
+
+    output:
+    tuple val(meta), path("pydamage_results/pydamage_results.csv"), path("pydamage_results/pydamage_filtered_results.csv"), emit: csv
```
Comment (jfy133) [inline modules/pydamage/main.nf:26]: Should be one channel = one output file

File: modules/pydamage/main.nf [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        $options.args \\
+        -p $task.cpus \\
+        $bam
+
+    pydamage \\
+        filter \\
+        pydamage_results/pydamage_results.csv
+
+
+    pydamage --version | sed -e 's/pydamage, version //g' > ${software}.version.txt
```
Comment (jfy133) [inline modules/pydamage/main.nf:44]: We apparently have to wrap the `<tool> --version` in a `echo \$(tool -version)` a lá: https://github.com/nf-core/modules/blob/6ff995e93ddec11d8c3112edcee231ce6777a15d/modules/bowtie2/align/main.nf#L48

File: modules/pydamage/main.nf [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    """
+    pydamage \\
+        analyze \\
+        $options.args \\
+        -p $task.cpus \\
+        $bam
+
+    pydamage \\
+        filter \\
+        pydamage_results/pydamage_results.csv
```
Comment (jfy133) [inline modules/pydamage/main.nf:41]:
```suggestion
    pydamage \\
        filter \\
        pydamage_results/pydamage_results.csv
```

This should go in a separate module

File: modules/pydamage/meta.yml [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+  - meta:
+      type: map
+      description: |
+        Groovy Map containing sample information
+        e.g. [ id:'test', single_end:false ]
+  - version:
+      type: file
+      description: File containing software version
+      pattern: "*.{version.txt}"
+  ## TODO nf-core: Delete / customise this example output
```
Comment (jfy133) [inline modules/pydamage/meta.yml:40]:
```suggestion
```

File: modules/pydamage/meta.yml [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+  - csv:
+      type: file
+      description: PyDamage results as csv files
+      pattern: "*.csv"
```
Comment (jfy133) [inline modules/pydamage/meta.yml:44]: Specify each output file (see comment above)

**Review by jfy133 — CHANGES_REQUESTED** [review]
mostly good, some 'guidelines' things that need to be tweaked

The major thing is to separate `analyze` from `filter` into two separate modules (should be completely atomic)

File: modules/pydamage/main.nf (hunk 1 above)
Comment (maxibor) [reply modules/pydamage/main.nf]: fixed

File: modules/pydamage/main.nf (hunk 3 above)
Comment (maxibor) [reply modules/pydamage/main.nf]: changed

File: modules/pydamage/main.nf (hunk 4 above)
Comment (maxibor) [reply modules/pydamage/main.nf]: done

File: modules/pydamage/analyze/main.nf [hunk 7]
```diff
@@ -0,0 +1,41 @@
+// Import generic module functions
+include { initOptions; saveFiles; getSoftwareName } from './functions'
+
+
```
Comment (jfy133) [inline modules/pydamage/analyze/main.nf:4]:
```suggestion
```

**Review by jfy133 — COMMENTED** [review]
Almost there, just formatting stuff now

**Review by jfy133 — APPROVED** [review]

**Pull request merged**